frontend/test/views/AnalysisView.spec.js

```diff
@@ -106,40 +111,107 @@ describe('AnalysisView', () => {
       );
     });
 
-    it('should mark an analysis as ready', async () => {
-      const annotatingAnalysis = fixtureData();
-      annotatingAnalysis.latest_status = 'Annotation';
-      mockedData.returns(annotatingAnalysis);
-      const wrapper = getMountedComponent();
-      await wrapper.vm.$nextTick();
-      const headerComponent = wrapper.getComponent(
-          '[data-test=analysis-view-header]',
-      );
-      const actionsProps = headerComponent.props('actions');
-      for (const action of actionsProps) {
-        if (action.text === 'Mark Ready') {
-          action.operation();
+    describe('actions with toasts', () => {
+      /**
+       * Helper triggers an action based on the action text
+       * @param {VueWrapper} wrapper The Vue wrapper containing the component instance
+       * @param {string} actionText The text of the action to trigger
+       * @return {Promise} A promise that resolves when the action is triggered
+       */
+      async function triggerAction(wrapper, actionText) {
```

Comment:
It's fine to make functions to avoid having repeat code, but having it in the middle of the file makes it hard to read and find. These functions should be at the beginning of the test file where everything else is defined.

We want to avoid hiding helper functions deep within test files. This might prevent other sections within the file using the same code segment down the road as it is scoped within a specific `describe`. Keep this available across the whole file.